src/main/java/com/notably/commons/path/RelativePath.java

```diff
@@ -19,6 +21,7 @@
     private final List<String> components;
 
     private RelativePath(String relativePathString) {
+        requireNonNull(relativePathString);
```

Comment:
You should add this `Objects.requireNonNull` call everywhere you can! For instance, you should also add it in the `private RelativePath(List<String> components)` constructor below, and all the other methods in this class.

Reply:
will do, thanks!